pennylane_lightning/src/tests/TestingUtils.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+// Helper macro to check if an exception is thrown with the correct complete error message
+#define EXPECT_THROW_WITH_MESSAGE(stmt, etype, whatstring) EXPECT_THROW( \
```

Comment:
Added as per https://github.com/google/googletest/issues/952#issuecomment-521361666 and the thread there. Any merged features related to this were before the release version that we are pulling at the moment.

Reply:
Strange that this isn't part of gtest already. 🤔

Reply:
Something seems to have been added, but the activity was after the latest release of `gtest` available on GitHub. So I haven't really explored too much on what they have added.